pyaedt/application/Design.py

```diff
@@ -2261,12 +2263,58 @@ def import_dataset3d(self, filename, dsname=None):
 
         >>> oProject.AddDataset
         """
-        with open(filename, "r") as f:
-            lines = f.read().splitlines()
-        header = lines[0]
-        points = lines[1:]
+        index_of_dot = filename.rfind(".")
+        file_extension = filename[index_of_dot + 1 :]
+        xlist = []
+        ylist = []
+        zlist = []
+        vlist = []
 
-        header_list = header.split("\t")
+        if file_extension == "xlsx":
+            self.logger.warning("Warning: You need pandas installed for reading excel files")
+            try:
+                import pandas as pd
+            except ImportError:
+                self.logger.error(
+                    "Pandas is not installed. Either install pandas or save the file as .csv or .tab. " "Exiting."
+                )
+                return
+
+            df = pd.read_excel(filename)
+            header = str([df.columns[i] for i in range(len(df.columns))])
+            xlist = list((df.iloc[:, 0]).array)
+            ylist = list((df.iloc[:, 1]).array)
+            zlist = list((df.iloc[:, 2]).array)
+            vlist = list((df.iloc[:, 3]).array)
+            # Add code to create the x,y,z,v lists once pandas is included in the pyaedt requirements
+
+        elif file_extension == "csv":
+            import csv
+
+            with open(filename, "r", encoding=encoding) as my_file:
+                csv_reader = csv.reader(my_file, delimiter=",")
+                i = 0
+                for row in csv_reader:
+                    if i == 0:
+                        header = " ".join(row)
+                        i = 1
+                        continue
+                    xlist.append(float(row[0]))
+                    ylist.append(float(row[1]))
+                    zlist.append(float(row[2]))
+                    vlist.append(float(row[3]))
+            my_file.close()
+        elif file_extension == "tab":
+            with open(filename) as my_file:
+                lines = my_file.readlines()
+            header = lines[0]
+            for item in lines[1:]:
+                xlist.append(float(item.split()[0]))
+                ylist.append(float(item.split()[1]))
+                zlist.append(float(item.split()[2]))
+                vlist.append(float(item.split()[3]))
+            my_file.close()
```

Comment:
@zebanaqviWAT You don't need to close the file hear as you are using a `with open` statement that will take care of closing the file after reading its content (here with the `readlines()` method).
```suggestion
```